bindings/rust/src/bindings/mod.rs

```diff
@@ -95,23 +95,23 @@ pub fn hex_to_bytes(hex_str: &str) -> Result<Vec<u8>, Error> {
 
 /// Holds the parameters of a kzg trusted setup ceremony.
 impl KZGSettings {
-    /// Initializes a trusted setup from `FIELD_ELEMENTS_PER_BLOB` g1 points in
-    /// Lagrange form, 65 g2 points in monomial form, and
-    /// `FIELD_ELEMENTS_PER_BLOB` g1 points in monomial form.
+    /// Initializes a trusted setup from a flat array of `FIELD_ELEMENTS_PER_BLOB` G1 points in
+    /// Lagrange form, a flat array of 65 G2 points in monomial form, and a flat array of
+    /// `FIELD_ELEMENTS_PER_BLOB` G1 points in monomial form.
     pub fn load_trusted_setup(
-        g1_monomial_bytes: &[[u8; BYTES_PER_G1_POINT]],
-        g1_lagrange_bytes: &[[u8; BYTES_PER_G1_POINT]],
-        g2_monomial_bytes: &[[u8; BYTES_PER_G2_POINT]],
+        g1_monomial_bytes: &[u8],
+        g1_lagrange_bytes: &[u8],
+        g2_monomial_bytes: &[u8],
         precompute: usize,
     ) -> Result<Self, Error> {
-        if g1_monomial_bytes.len() != FIELD_ELEMENTS_PER_BLOB {
+        if g1_monomial_bytes.len() != BYTES_PER_BLOB {
             return Err(Error::InvalidTrustedSetup(format!(
                 "Invalid number of g1 monomial points in trusted setup. Expected {} got {}",
                 FIELD_ELEMENTS_PER_BLOB,
                 g1_monomial_bytes.len()
             )));
         }
-        if g1_lagrange_bytes.len() != FIELD_ELEMENTS_PER_BLOB {
+        if g1_lagrange_bytes.len() != BYTES_PER_BLOB {
```

Comment:
For these two checks, I think I would prefer to use `FIELD_ELEMENTS_PER_BLOB*BYTES_PER_G1_POINT` for clarity and consistency with the G2 points check.

Reply:
Fixed in c4260665b